src/include/rados/librados.hpp

```diff
@@ -1154,6 +1154,9 @@ namespace librados
 
     config_t cct();
 
+    void set_honor_osdmap_full();
+    void unset_honor_osdmap_full();
+
```

Comment:
It would still be *far* preferably to use the existing librados flags, though!  Please try that appraoch first.. hopefully there is a central place wehre that can be done, or add an ImageCtx 'rados_flags' variable that is |'d in to every request.